src/test/java/org/cactoos/map/MapEnvelopeTest.java

```diff
@@ -46,93 +43,73 @@
 @SuppressWarnings("PMD.TooManyMethods")
 public final class MapEnvelopeTest {
 
-    /**
-     * A rule for handling an exception.
-     */
-    @Rule
-    public final ExpectedException exception = ExpectedException.none();
-
     @Test
     public void putThrowsException() {
-        this.exception.expect(UnsupportedOperationException.class);
-        this.exception.expectMessage(
-            "#put() is not supported, it's a read-only map"
-        );
         MatcherAssert.assertThat(
-            "put method did not throw exception",
-            new NoNulls<>(
+            "put method must throw exception",
+            () -> new NoNulls<>(
                 new MapOf<Integer, Integer>(
-                    new MapEntry<Integer, Integer>(0, -1)
+                    new MapEntry<>(0, -1)
                 )
-            ),
-            new MatcherOf<>(
-                new FuncOf<>(
-                    (map) -> map.put(2, 2),
-                    true
-                ))
+            ).put(2, 2),
+            new Throws<>(
+                "#put() is not supported, it's a read-only map",
+                UnsupportedOperationException.class
+            )
         );
     }
 
     @Test
     public void removeThrowsException() {
-        this.exception.expect(UnsupportedOperationException.class);
-        this.exception.expectMessage(
-            "#remove() is not supported, it's a read-only map"
-        );
         MatcherAssert.assertThat(
             "remove method did not throw exception",
-            new NoNulls<>(
+            () -> new NoNulls<>(
                 new MapOf<Integer, Integer>(
                     new MapEntry<>(0, -1)
                 )
-            ),
-            new MatcherOf<>(
-                new FuncOf<>(
-                    (map) -> map.remove(0),
-                    true
-                ))
+            ).remove(0),
+            new Throws<>(
+                "#remove() is not supported, it's a read-only map",
+                UnsupportedOperationException.class
+            )
         );
     }
 
     @Test
     public void putAllThrowsException() {
-        this.exception.expect(UnsupportedOperationException.class);
-        this.exception.expectMessage(
-            "#putAll() is not supported, it's a read-only map"
-        );
         MatcherAssert.assertThat(
```

Comment:
@atapin please use `Assertion` instead of the static methods from `MatcherAssert`